src/unit_tests/wazuh_modules/vulnerability_detector/test_wm_vuln_detector.c

```diff
@@ -18609,6 +18610,285 @@ void test_wm_vuldet_send_removed_cve_report_no_ip_success() {
+    will_return(__wrap_cJSON_ParseWithOpts, j_object);
+
+    will_return(__wrap_cJSON_GetObjectItem, j_status);
+    will_return_count(__wrap_cJSON_GetStringValue, "SUCCESS", 1);
+    expect_function_calls(__wrap_cJSON_Delete, 1);
+
+    result = wm_vuldet_recv_wdb_by_row(&array);
+
+    __real_cJSON_Delete(array);
+    __real_cJSON_Delete(j_status);
+    __real_cJSON_Delete(j_object);
+    assert_int_equal(result, OS_SUCCESS);
+}
+
+void test_wm_vuldet_recv_wdb_by_row_no_status() {
+    int result;
+    char* response = "ok {}";
+    cJSON* array = __real_cJSON_CreateArray();
+    cJSON* j_object = __real_cJSON_Parse(response + 3);
+
+    wdb_vuldet_sock = 1;
+
+    will_return(__wrap_cJSON_CreateArray, array);
+
+    expect_value(__wrap_OS_RecvSecureTCP, sock, 1);
+    expect_value(__wrap_OS_RecvSecureTCP, size, OS_MAXSTR);
+    will_return(__wrap_OS_RecvSecureTCP, response);
+    will_return(__wrap_OS_RecvSecureTCP, strlen(response));
+
+    will_return(__wrap_cJSON_ParseWithOpts, j_object);
+
+    will_return(__wrap_cJSON_GetObjectItem, NULL);
+    expect_function_calls(__wrap_cJSON_Delete, 2);
+
+    result = wm_vuldet_recv_wdb_by_row(&array);
+
+    __real_cJSON_Delete(array);
+    __real_cJSON_Delete(j_object);
+    assert_int_equal(result, OS_INVALID);
+}
+
+void test_wm_vuldet_recv_wdb_by_row_success() {
+    int result;
+    char* response = "due {\"name\":\"package1\"}";
+    char* response2 = "ok {\"status\":\"SUCCESS\"}";
+    cJSON* array = __real_cJSON_CreateArray();
+    cJSON* j_status = __real_cJSON_CreateString("SUCCESS");
+    cJSON* j_object = __real_cJSON_Parse(response + 4);
+    cJSON* j_object2 = __real_cJSON_Parse(response2 + 3);
+
+    wdb_vuldet_sock = 1;
+
+    will_return(__wrap_cJSON_CreateArray, array);
+
+    // First message
+    expect_value(__wrap_OS_RecvSecureTCP, sock, 1);
+    expect_value(__wrap_OS_RecvSecureTCP, size, OS_MAXSTR);
+    will_return(__wrap_OS_RecvSecureTCP, response);
+    will_return(__wrap_OS_RecvSecureTCP, strlen(response));
+    will_return(__wrap_cJSON_ParseWithOpts, j_object);
+
+    expect_function_call(__wrap_cJSON_AddItemToArray);
+    will_return(__wrap_cJSON_AddItemToArray, TRUE);
+
+    // Final message
+    expect_value(__wrap_OS_RecvSecureTCP, sock, 1);
+    expect_value(__wrap_OS_RecvSecureTCP, size, OS_MAXSTR);
+    will_return(__wrap_OS_RecvSecureTCP, response2);
+    will_return(__wrap_OS_RecvSecureTCP, strlen(response2));
+    will_return(__wrap_cJSON_ParseWithOpts, j_object2);
+
+    will_return(__wrap_cJSON_GetObjectItem, j_status);
+    will_return_count(__wrap_cJSON_GetStringValue, "SUCCESS", 1);
+    expect_function_calls(__wrap_cJSON_Delete, 1);
+
+    result = wm_vuldet_recv_wdb_by_row(&array);
+
+    __real_cJSON_Delete(array);
+    __real_cJSON_Delete(j_status);
+    __real_cJSON_Delete(j_object);
+    __real_cJSON_Delete(j_object2);
+    assert_int_equal(result, OS_SUCCESS);
+}
+
+// Tests wm_vuldet_get_hotfixes
+
+void test_wm_vuldet_get_hotfixes_send_error() {
```

Comment:
I think we should add the success case for these UTs. It will be too similar to any of the **test_wm_vuldet_recv_wdb_by_row_success**, but, it will be correct to include it and will cover 100% of coverage.

Reply:
Fixed